book-service/src/main/java/com/jpop8/bookservice/services/BookService.java

```diff
@@ -0,0 +1,47 @@
+package com.jpop8.bookservice.services;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.stereotype.Service;
+
+import com.jpop8.bookservice.exceptions.ResourceNotFoundException;
+import com.jpop8.bookservice.models.Book;
+import com.jpop8.bookservice.repository.BookCategoryRepository;
+import com.jpop8.bookservice.repository.BookRepository;
+
+@Service
+public class BookService {
+	private BookRepository bookRepository;
```

Comment:
@Autowired

Reply:
I remember reading an article long time back related to @Autowired being optional in case there's only one candidate for injection. I found a similar matching argument here: https://stackoverflow.com/a/68544892
Let me know what you think!

Reply:
Anyways even we add @Autowired here, we are not narrowing anything down. It improves readabilty,
Let it be now. Thanks for sharing links